sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/data/SpanData.scala

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk
+package trace
+package data
+
+import cats.Hash
+import cats.Show
+import cats.syntax.foldable._
+import cats.syntax.show._
+import org.typelevel.otel4s.sdk.common.InstrumentationScope
+import org.typelevel.otel4s.trace.SpanContext
+import org.typelevel.otel4s.trace.SpanKind
+
+import scala.concurrent.duration.FiniteDuration
+
+/** Immutable representation of all data collected by the
+  * [[org.typelevel.otel4s.trace.Span Span]].
+  *
+  * @see
+  *   [[https://opentelemetry.io/docs/specs/otel/trace/api/#span]]
+  */
+sealed trait SpanData {
+
+  /** The name of the span.
+    */
+  def name: String
+
+  /** The span context associated with this span.
+    */
+  def spanContext: SpanContext
+
+  /** The parent span context, if any.
+    */
+  def parentSpanContext: Option[SpanContext]
+
+  /** The kind of the span.
+    */
+  def kind: SpanKind
+
+  /** The start timestamp of the span.
+    */
+  def startTimestamp: FiniteDuration
+
+  /** The end time of the span.
+    */
+  def endTimestamp: Option[FiniteDuration]
```

Comment:
We can use shorter alternatives:

```scala
def start: FiniteDuration
def end: Option[FiniteDuration]
```

WDYT?

Reply:
This isn't really user-facing tho right? I think I prefer the current names.